### DISCUSSION TITLE
[Combobox] Prop for clearing input on click

### DISCUSSION THREAD

**Discussion by alexbu92** [tailwindlabs/headlessui#2014] [Ideas]
as per title, a prop that if enabled, automatically clears the input of the displayValue when a user clicks on the combobox to open it.

Currently, I haven't yet found a way to clear it, so if a user wants to search for something they first need to remove all characters in the input.

**Comment by vincentbernat** [comment]
Previously, the input was cleared on click. I don't find this change in the changelog and looking at the history of commits on the component, I don't find what could have caused this.

**Comment by schmoli** [comment]
Trying to do this right now as well, any luck?

**Comment by alexbu92** [comment]
No, not currently possible.

**Comment by sebsobseb** [comment]
Another approach that can be considered is to **select all text** when clicking in the input field. Which has a better UX imo.

**Reply by camiaei** [reply]
hi! it seems that there is still no way to do it 😔

> as per title, a prop that if enabled, automatically clears the input of the displayValue when a user clicks on the combobox to open it. 

> Another approach that can be considered is to **select all text** when clicking in the input field. Which has a better UX imo.

either approach would be good 🙏

**Comment by RIP21** [comment]
Having ability to clear or select all text would be awesome as for now I cannot hack around the selection of the text with refs and whatnot it doesn't work. E.g. `select()` on the `input` doesn't work :(
